### PR TITLE
Feature/deterministic safety test

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt/src/integration/java/com/radixdlt/integration/invariants/SafetyChecker.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			.reduce((v0, v1) -> v0.compareTo(v1) < 0 ? v0 : v1)
+			.orElse(EpochView.of(0, View.genesis()));
+		committedVertices.headMap(lowest).clear();
+
+		return Optional.empty();
+	}
+
+	public Optional<TestInvariantError> process(BFTNode node, BFTCommittedUpdate committedUpdate) {
+		ImmutableList<PreparedVertex> vertices = committedUpdate.getCommitted();
+		for (PreparedVertex vertex : vertices) {
```
Comment (siy) [inline radixdlt/src/integration/java/com/radixdlt/integration/invariants/SafetyChecker.java:124]: Alternative variant:
```
return vertices.stream()
		.map(vertex -> process(node, vertex.getVertex()))
		.filter(Optional::isPresent)
		.map(Optional::get)
		.findFirst();
```

**Review by siy — APPROVED** [review]
LGTM with minor comment

File: radixdlt/src/integration/java/com/radixdlt/integration/recovery/OneNodeAlwaysAliveSafetyTest.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+			try {
+				injector.getInstance(DeterministicEpochsConsensusProcessor.class).handleMessage(msg.value().origin(), msg.value().message());
+			} finally {
+				ThreadContext.remove("bftNode");
+			}
+		}
+	}
+
+	@Test
+	@Ignore("Safety broken here.")
```
Comment (clstrfsck) [inline radixdlt/src/integration/java/com/radixdlt/integration/recovery/OneNodeAlwaysAliveSafetyTest.java:225]: Can we get an issue in Jira, and a reference to that issue in the comment here please.

**Review by clstrfsck — APPROVED** [review]
Looks OK.

**Pull request merged**